docker/local/Dockerfile.local

```diff
@@ -75,18 +81,50 @@ RUN TORCH_MAJOR_MINOR=$(echo $TORCH_VERSION | cut -c1-3) && \
     transformers==${TRANSFORMERS_VERSION} \
     && rm -rf /root/.cache
 
-FROM base-${DEVICE} as runtime
+FROM base-${DEVICE} as parmetis-true
 
-ENV PYTHONPATH="/root/dgl/tools/:${PYTHONPATH}"
+# Install MPI and dependencies
+RUN apt update && apt install -y --no-install-recommends \
+    build-essential \
+    cmake \
+    libopenmpi-dev \
+    openmpi-bin \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN pip install \
+    pyyaml \
+    && rm -rf /root/.cache
 
-# Download DGL source code
-RUN cd /root; git clone --branch v${DGL_VERSION} https://github.com/dmlc/dgl.git
+# Install GKLib
+RUN cd /root && \
+    git clone --single-branch --branch master https://github.com/KarypisLab/GKlib && \
+    cd GKlib && \
+    make && \
+    make install
 
-# Copy GraphStorm source and add to PYTHONPATH
-RUN mkdir -p /graphstorm
-COPY code/python/graphstorm /graphstorm/python/graphstorm
-ENV PYTHONPATH="/graphstorm/python/:${PYTHONPATH}"
+# Install Metis
+RUN cd /root && \
+    git clone --single-branch --branch master https://github.com/KarypisLab/METIS.git && \
+    cd METIS && \
+    make config shared=1 cc=gcc prefix=/root/local i64=1 && \
+    make install
+
+# Install Parmetis
+RUN cd /root && \
+    git clone --single-branch --branch main https://github.com/KarypisLab/PM4GNN.git && \
+    cd PM4GNN && \
+    make config cc=mpicc prefix=/root/local && \
+    make install
 
+ENV PATH=$PATH:/root/local/bin
+ENV LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/root/local/lib/
+RUN cp /root/local/bin/pm_dglpart /root/local/bin/pm_dglpart3
```

Comment:
Why do this?

Reply:
I think this is needed for the parmetis executable and portable library to be available to bash and any running process. From https://github.com/KarypisLab/ParMETIS?tab=readme-ov-file#building-and-installing-parmetis